### PR TITLE
add id to raw msg view

### PR DESCRIPTION
**Pull request by mixmix**
selection_259

### CODE REVIEW

**Comment by mixmix** [comment]
ooo found a bug when trying to post a message ..

File: ui/com/pretty-raw.js [hunk 1]
```diff
@@ -36,11 +38,13 @@ function elements(path, obj, Com) {
...
   constructor(props) {
     super(props)
-    this.state = { obj: null }
+    this.state = { obj: null, key: null }
   }
   componentDidMount() {
     var obj = this.props.obj
-    this.setState({ obj: obj })
+    var id = this.props.id
+    console.log(this.props)
```
Comment (pfraze) [inline ui/com/pretty-raw.js]: leftover logging

**Comment by cryptix** [comment]
Hrm... It looks like ID is part of the post now which is different from this:

```
$ sbot get "%4iHjk8gY5rqBY8BJX9xxLA5yptCrp7+ctpiZYmXMld0=.sha256"
{
  "previous": "%B6piVpVCyQM/04HVpdL8WgljE2YBolI69PmqrXBn2A8=.sha256",
  "author": "@ye+QM09iPcDJD6YvQYjoQc7sLF/IFhmNbEqgdzQo3lQ=.ed25519",
  "sequence": 1882,
  "timestamp": 1455476705706,
  "hash": "sha256",
  "content": {
    "type": "post",
    "text": "That should be fairly simple :+1: ",
    "root": "%Vl5PPKI7n3QZClQmlLdiw1V7JFCKBuz7nEpYBCFztcU=.sha256",
    "branch": "%Vl5PPKI7n3QZClQmlLdiw1V7JFCKBuz7nEpYBCFztcU=.sha256",
    "channel": "patchwork-dev"
  },
  "signature": "KsoT924bIbW16EjJuWuoiVgWqv0LRKG/JTMr29eJt3Dj4a1BNFj7JsSFdzrI0BG3iERUy3VHgfFRGgJTAdGwBA==.sig.ed25519"
}
```

I'm quite certain that the orig question here was something like "is this content addressable or what?" so the ID might fit better over the tabular view.

File: ui/com/pretty-raw.js (hunk 1 above)
Comment (mixmix) [inline ui/com/pretty-raw.js]: derp. thanks

**Comment by mixmix** [comment]
@cryptix yeah looks like we could pass the whole message around. Happy to refactor that way if @pfraze thinks it's a good idea but I wasn't sure if we wanted to teach Table about `msg` anatomy

**Comment by mixmix** [comment]
mind you I might have broken that rule by treating id as some special prop

**Comment by pfraze** [comment]
Yeah, one option would be to leave the pretty-raw components untouched and change msg-content's raw rendering to something like:

```
<pre><code>{id+'\n'+JSON.stringify(msg.value,null, 2)}</code></pre>
```

So that we get something similar to what @cryptix just posted above (minus the CLI facets).

**Comment by mixmix** [comment]
Here's the alternative : https://github.com/ssbc/patchwork/pull/302

**Comment by mixmix** [comment]
Wait doesn't this pull request get out moded by that raw json one @pfraze
?

Oh i remember undying a bunch of the changes in the other pr.  Hope you
merged this one first!

**Comment by pfraze** [comment]
Yeah if #302 was a branch of #300, then merging #302 effectively merged this one too

**Comment by pfraze** [comment]
Magics!

**Pull request merged**